src/v/cloud_storage_clients/abs_client.h

```diff
@@ -279,6 +286,13 @@ class abs_client : public client {
     ss::future<storage_account_info>
     do_get_account_info(ss::lowres_clock::duration timeout);
 
+    /// \brief test if Hierarchical Namespace is enabled in the storage account
+    /// by creating a test file and calling Set Expiry on this. Set Expiry is
+    /// available only when HNS is enabled. use the result to infer if HNS is
+    /// enabled.
+    ss::future<storage_account_info>
+    do_test_set_expiry_on_dummy_file(ss::lowres_clock::duration timeout);
```

Comment:
This method is called `do_test_set_expiry_on_dummy_file` but it returns `storage_account_info`. That just doesn't make sense. Why not make it return a bool?

Reply:
it's an alternative for 
`do_get_account_info()` and it returns the same type. 
maybe I can rename the caller to `get_storage_account_info()` , and the two functions to something like
`do_storage_account_info_via_get_account_info` and `do_storage_account_info_via_set_expiry` ? less confusing?

Reply:
It is not an alternative as it won't ever return account information. We just infer one property of the account. Instead of _upgrading_ the "is hns supported test" to "account info" I suggest _downgrading_ the existing "account info" method to something like "is hns supported" too.